src/server/lib/withStorage.js

```diff
@@ -2,6 +2,46 @@
  * @param {import("../../data/types.js").App} App
  */
 export function withStorage(App) {
+  // Simple in-memory cache with LRU eviction
+  const cache = new Map()
+  const maxCacheSize = 1000 // Limit cache size to prevent memory issues
+
+  /**
+   * @param {string} key
+   * @param {string} value
+   */
+  function setCacheItem(key, value) {
+    // If cache is at max size, delete oldest entry (first in map)
+    if (cache.size >= maxCacheSize && !cache.has(key)) {
```

Comment:
question: what about performance? Isn't it slow to remove one item at a time?

But anyways, increase the max size to 10k